feat: add Kotlin for Data Analysis section (#4125)
JetBrains/kotlin-web-site: docs/topics/data-analysis/data-science-libraries.md

|--------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | [Kotlin DataFrame](https://github.com/Kotlin/dataframe)                  | - Data collection <br/> - Data cleaning and processing                                                                   | - Operations for creating, sorting, and cleaning data frames, feature engineering, and more  <br/> - Processing of structured data  <br/> - Support for `CSV`, `JSON`, and other input formats <br/> - Reading from SQL databases <br/> - Connecting with different APIs to access data and increase type safety                                                                                                                                                                                                                                                                                                                                                          |

Reviewer: ```suggestion
| [Kotlin DataFrame](https://github.com/Kotlin/dataframe)                  | - Data collection <br/> - Data cleaning and processing                                                                   | - Operations for creating, sorting, and cleaning data frames, feature engineering, and more  <br/> - Processing of structured data  <br/> - Support for CSV, JSON, and other input formats <br/> - Reading from SQL databases <br/> - Connecting with different APIs to access data and increase type safety                                                                                                                                                                                                                                                                                                                                                          |
```

> | [Kotlin DataFrame](https://github.com/Kotlin/dataframe)                  | - Data collection <br/> - Data cleaning and processing                                                                   | - Operations for creating, sorting, and cleaning data frames, feature engineering, and more  <br/> - Processing of structured data  <br/> - Support for `CSV`, `JSON`, and other input formats <br/> - Reading from SQL databases <br/> - Connecting with different APIs to access data and increase type safety                                                                                                                                                                                                                                                                                                                                                          |

Reviewer: ```suggestion
| [Kotlin DataFrame](https://kotlin.github.io/dataframe/overview.html)                  | - Data collection <br/> - Data cleaning and processing                                                                   | - Operations for creating, sorting, and cleaning data frames, feature engineering, and more  <br/> - Processing of structured data  <br/> - Support for `CSV`, `JSON`, and other input formats <br/> - Reading from SQL databases <br/> - Connecting with different APIs to access data and increase type safety                                                                                                                                                                                                                                                                                                                                                          |
```

| [Smile](https://github.com/haifengl/smile)                                    | - Data cleaning and processing <br/> - Data exploration and visualization <br/> - Model building  | - Ready-made algorithms for machine learning and natural language processing <br/> - Linear algebra, graph, interpolation, and visualization tools <br/> - Provides functional [Kotlin API](https://github.com/haifengl/smile/tree/master/kotlin), [Scala API](https://github.com/haifengl/smile/tree/master/scala), [Clojure API](https://github.com/haifengl/smile/tree/master/clojure), and more |
| [Smile-NLP-kt](https://github.com/londogard/smile-nlp-kt)                     | Data cleaning and processing                                                                      | - Kotlin rewrite of the [Scala](https://www.scala-lang.org/api/current/) implicits for the natural language processing part of Smile <br/> - Operations in the format of Kotlin extension functions and interfaces <br/> - Sentence breaking, stemming, bag of words, and other tasks                                                                                                               |

> | [Weka](https://www.cs.waikato.ac.nz/ml/index.html)                            | - Data cleaning and processing <br/> - Data exploration and visualization  <br/> - Model building | - Collection of machine learning algorithms for data mining tasks   <br/> - Tools for data preparation, classification, regression, clustering, association rules mining, and visualization                                                                                                                                                                                                         |

Reviewer: ```suggestion
```
